benchmark/training/training_benchmark.py

```diff
@@ -0,0 +1,189 @@
+                for layers in args.num_layers:
+                    num_neighbors = neighbors_sizes
+                    if type(neighbors_sizes) is list:
+                        if len(neighbors_sizes) == 1:
+                            num_neighbors = neighbors_sizes * layers
+                    elif type(neighbors_sizes) is int:
+                        num_neighbors = [neighbors_sizes] * layers
+
+                    assert len(
+                        num_neighbors) == layers, \
+                        f'''num_neighbors={num_neighbors} lenght
+                        != num of layers={layers}'''
+
+                    subgraph_loader = NeighborLoader(
+                        data,
+                        num_neighbors=num_neighbors,
+                        input_nodes=mask,
+                        batch_size=batch_size,
+                        shuffle=True,
+                        num_workers=args.num_workers,
+                    )
+                    for hidden_channels in args.num_hidden_channels:
+                        print('----------------------------------------------')
+                        print(f'Batch size={batch_size}, '
+                              f'Layers amount={layers}, '
+                              f'Num_neighbors={num_neighbors}, '
+                              f'Hidden features size={hidden_channels}, '
+                              f'Sparse tensor={args.use_sparse_tensor}')
+
+                        params = {
+                            'inputs_channels': inputs_channels,
+                            'hidden_channels': hidden_channels,
+                            'output_channels': num_classes,
+                            'num_heads': args.num_heads,
+                            'num_layers': layers,
+                        }
+
+                        if model_name == 'pna':
+                            if degree is None:
+                                degree = PNAConv.get_degree_histogram(
+                                    subgraph_loader)
+                                print(f'Calculated degree for {dataset_name}.')
+                            params['degree'] = degree
+
+                        model = get_model(
+                            model_name, params,
+                            metadata=data.metadata() if hetero else None)
+                        model = model.to(device)
+                        model.train()
+                        optimizer = torch.optim.Adam(model.parameters(),
+                                                     lr=0.001)
+
+                        progress_bar = False if args.no_progress_bar else True
+                        with amp:
+                            for _ in range(args.warmup):
+                                train(model, subgraph_loader, optimizer,
+                                      device, hetero,
+                                      progress_bar=progress_bar, desc="Warmup")
+                            with timeit():
```

Comment:
Recording the `num_pochs` epochs is too heavy for benchmark. We can only record the last epoch of training. Same as the following profiling part.